src/stores/likes.js

```diff
@@ -33,6 +36,128 @@ export const useLikeStore = defineStore('likes', {
       }
     },
 
+    /**
+     *
+     * @param promptId
+     * @param startAt Start Date
+     * @returns {Promise<void>}
+     */
+    async fetchPromptStat(promptId, startAt) {
+      let endAt = Timestamp.now()
+      if (endAt - startAt > 2678400) {
+        endAt = new Timestamp(startAt.seconds + 2678400, 0)
+      }
+      const _calendarDay = calendarDay(startAt, endAt)
+      const _calendarWeek = calendarWeek(startAt, endAt)
+
+      const fetchCalendarData = async (calendar) => {
+        const stats = []
+        for (let i = 0; i < calendar.length; i++) {
+          const date = calendar[i]
+          let nextDate
+          if (i + 1 < calendar.length) {
+            nextDate = calendar[i + 1]
+          } else {
+            nextDate = Timestamp.fromDate(new Date())
+          }
+          const likesCollection = collection(db, 'prompts', promptId, 'likes')
+          const dislikesCollection = collection(db, 'prompts', promptId, 'dislikes')
+          const likesQuery_ = query(
+            likesCollection,
+            where('createdAt', '>=', date.toDate().getTime()),
+            where('createdAt', '<', nextDate.toDate().getTime())
+          )
+          const likeSnapshot = await getCountFromServer(likesQuery_)
+
+          const dislikesQuery_ = query(
+            dislikesCollection,
+            where('createdAt', '>=', date.toDate().getTime()),
+            where('createdAt', '<', nextDate.toDate().getTime())
+          )
+          const dislikesSnapshot = await getCountFromServer(dislikesQuery_)
```

Comment:
Avoid calling APIs inside loops.
This can slow down the application or "load forever" when calling `getCountFromServer()` inside a loop.

Instead, try calling all documents and then applying logic to split days, weeks, months, etc.

This will prevent overcharging and your function will be shorter.